upgrade/upgrade-scripts/verify-impl

```diff
@@ -36,6 +37,94 @@ function run_support_info() {
 	return "$rc"
 }
 
+function verify_run_verify_jar() {
+	echo "Verify - Run verify jar."
+
+	/usr/bin/java \
+		-Dlog.dir=/var/delphix/server/upgrade-verify \
+		-Dmdsverify=true \
+		"${VERIFY_DEBUG_OPT}" \
+		-jar /opt/delphix/server/lib/exec/upgrade-verify/upgrade-verify.jar \
+		-d "${opt_o:-${LOG_DIRECTORY}/${CONTAINER}/upgrade_verify.json}" \
+		-f "${opt_f:-1}" \
+		-l "${opt_l:-en-US}" \
+		-v "$IMAGE_VERSION" \
+		-pl 60 -ph 80 \
+		"${VERIFY_LIVE_MDS_OPT}" ||
+		die "'upgrade-verify.jar' failed in verification container"
+
+	#
+	# This name is used by the "upgrade-verify.jar" just executed, so we
+	# cannot change this value without also modifying that JAR.
+	#
+	MDS_SNAPNAME="MDS-CLONE-upgradeverify"
+
+	# TODO : Make sure MDS clone gets cleaned up if verification fails.
+	/opt/delphix/server/bin/dx_manage_pg stop -s "$MDS_SNAPNAME" ||
+		die "failed to stop postgres"
+
+	/opt/delphix/server/bin/dx_manage_pg cleanup -s "$MDS_SNAPNAME" ||
+		die "failed to cleanup postgres"
+}
+
+function verify_stack_startup() {
+	echo "Verify - Run stack startup."
+
+	#
+	# Make sure delphix-mgmt and delphix-postgres aren't running within container.
+	#
+	[[ $(systemctl is-active delphix-mgmt) == 'inactive' ]] ||
+		die "delphix-mgmt service is not inactive."
+	[[ $(systemctl is-active delphix-postgres@default.service) == 'inactive' ]] ||
+		die "delphix-postgres@default.service service is not inactive."
+
+	#
+	# Create a copy of MDS and start the postgres service.
+	#
+	STACK_STARTUP_MDS_SNAPNAME="MDS-CLONE-stack-startup-verify"
+	/opt/delphix/server/bin/dx_manage_pg clone -s "${STACK_STARTUP_MDS_SNAPNAME}" ||
+		die "failed to clone postgres"
+	/opt/delphix/server/bin/dx_manage_pg start -s "${STACK_STARTUP_MDS_SNAPNAME}" -p 5432 ||
+		die "failed to start postgres"
+
+	#
+	# Modify application stack files to point to copy of MDS.
```

Comment:
Can you also leave a comment about why we need to do this?

Could we encapsulate this logic into a script delivered by the app-gate? At first glance it seems like very specific to the app-gate internals, so it'd be nice if we can avoid leaking these details into this script.. at the same time, this script is already very specific to the app-gate, so it's not that big of a deal.

Reply:
Sounds good to me. I'll move these logic into `dx_manage_pg.sh` script.

Reply:
Logic is removed to `dx_manage_pg.sh` script. It'll now take a extra `-l` parameter with `clone` command to replace symlinks after it clones MDS.